CMakeLists.txt

```diff
@@ -32,8 +32,7 @@ set(HIP_SUPPORTED_ARCHS "gfx906;gfx908;gfx90a;gfx940;gfx941;gfx942;gfx1030;gfx11
 # versions are derived from Dockerfile.rocm
 #
 set(TORCH_SUPPORTED_VERSION_CUDA "2.3.0")
-set(TORCH_SUPPORTED_VERSION_ROCM_5X "2.0.1")
-set(TORCH_SUPPORTED_VERSION_ROCM_6X "2.1.1")
+set(TORCH_SUPPORTED_VERSION_ROCM "2.4.0")
```

Comment:
QQ: How stable is PyTorch+ROCm 2.4.0?

Reply:
We've been using various 2.4.0 pins internally for a while and so far CI has not flagged any issues arising from PyTorch 2.4.0. I've pinned the specific PyTorch 2.4.0 wheel that we install to a known good version. We need PyTorch 2.4.0 because this PR https://github.com/vllm-project/vllm/pull/5473 introduced stateless device counts which are supported on ROCm only from 2.4.0 onwards.